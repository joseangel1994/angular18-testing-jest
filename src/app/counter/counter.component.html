<h2>Counter {{ counter }}</h2>
<button (click)="increaseBy(1)">+1</button>
<button (click)="increaseBy(-1)">-1</button>
