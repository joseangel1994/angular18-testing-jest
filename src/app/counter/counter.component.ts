import { Component } from '@angular/core';

@Component({
  selector: 'app-counter',
  standalone: true,
  imports: [],
  templateUrl: './counter.component.html'
})
export class CounterComponent {
  counter: number = 0;

  increaseBy(value: number) {
    this.counter += value;
  }
}
