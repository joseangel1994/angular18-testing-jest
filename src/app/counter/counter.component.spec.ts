import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from './counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [CounterComponent]
    })
    .compileComponents();

    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(compiled).toMatchSnapshot();
  })

  it('increaseBy should increment according to the argument', () => {
    component.increaseBy(1);
    expect(component.counter).toBe(1);
  })

  it('click on buttons should increment and decrement', () => {
    const buttons = Array.from(compiled.querySelectorAll('button'))
    const [increaseButton, decreaseButton] = buttons

    increaseButton.click()
    expect(component.counter).toBe(1)

    decreaseButton.click()
    expect(component.counter).toBe(0)
  })

  it('update counter should update template', () => {
    component.increaseBy(5)
    fixture.detectChanges()

    const h2 = compiled.querySelector('h2')
    expect(h2?.textContent).toContain('5')
  })
});
