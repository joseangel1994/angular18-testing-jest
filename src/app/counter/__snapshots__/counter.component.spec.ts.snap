// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <h2>
    Counter 0
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
</div>
`;
